pallets/xyk/src/lib.rs

```diff
@@ -864,6 +915,29 @@ impl<T: Config> Pallet<T> {
 			(current_time, pool_work_total, pool_missing_at_checkpoint),
 		);
 
+		LiquidityMiningActiveUser::<T>::try_mutate((&user, liquidity_asset_id), |active_amount| {
+			if let Some(val) = active_amount.checked_add(liquidity_assets_added) {
+				*active_amount = val;
+				Ok(())
+			} else {
+				Err(())
+			}
+		});
```

Comment:
If this fails, the rest will continue to execute, right?
That doesn't seem safe.

Reply:
added checked error:

`LiquidityMiningActivePool::<T>::try_mutate(liquidity_asset_id, |active_amount| {
			if let Some(val) = active_amount.checked_add(liquidity_assets_added) {
				*active_amount = val;
				Ok(())
			} else {
				Err(())
			}
		})
		.map_err(|_| DispatchError::from(Error::<T>::MathOverflow))?;`